src/main/webapp/user_script.js

```diff
@@ -34,34 +34,61 @@ function loadProfile() {
   });
 }
 
+/**Fetches the items that were liked by user and sends the array to 
+* populate the page with the item cards.
+*/
 function loadFavItems() {
   fetch('/favorite-item')
     .then((response) => response.json())
     .then((favorites) => {
       var itemIdArray = new Array();
-      const itemId = favorites;
-      itemIdArray.push(itemId);
-      itemIdArray.forEach(getItem);
+      itemIdArray = favorites;
+      populateFavoriteItemGrid(itemIdArray)
     })
     .catch((error) => {
-        console.log('Fetching favorite item data servlet failed: ' + error);
+      console.log('Fetching favorite item data servlet failed: ' + error);
     });
 }
 
-function getItem(item) {
-  if (item !== '') {
-    fetch(`/itempagedata?itemId=${item}`)
-      .then((response) => response.json())
-      .then((itemPageData) => {
-        createSelectedItemCard(itemPageData.item);
-      })
-      .catch((error) => {
-        console.log('Fetching favorite item data servlet failed: ' + error);
-      });
+/**
+ * Creates and populates a grid with item cards.
+ *
+ * @param { Array } favoriteItems - the array of item ids
+ */
+function populateFavoriteItemGrid(favoriteItems) {
+  const favItemContainer = $('#item-container');
+  let currItemIndex = 0;
+
+  while (currItemIndex < favoriteItems.length) {
+ 
+    const rowElem = $('<div class="row"></div>');
+
+    const MAX_CELLS_PER_ROW = 3;
```

Comment:
this constant can be defined outside the while loop (line 61) since its value never changes.